### PR TITLE
[TESB-19484] tJMSInput and tJMSOutput do not handle exceptions correctly

### PR DESCRIPTION
**Pull request by vbogatyrov**

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tJMSInput/tJMSInput_finally.javajet [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+CodeGeneratorArgument codeGenArgument = (CodeGeneratorArgument) argument;
+INode node = (INode)codeGenArgument.getArgument();
+String cid = node.getUniqueName();
+log = new LogUtil(node);
+String timeout=ElementParameterParser.getValue(node, "__TIMEOUT__");
+String maxMsg=ElementParameterParser.getValue(node, "__MAX_MSG__");
+%>
+if(globalMap.containsKey("consumer_<%=cid %>")) try {
+	((javax.jms.MessageConsumer)globalMap.get("consumer_<%=cid %>")).close();
+} catch (Exception e){
```
Comment (wwang-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tJMSInput/tJMSInput_finally.javajet]: please add the cid for the var e : e_<%=cid%>

**Comment by pteyssier** [comment]
Was it working in parallel exec before? because from the code it's not being more thread-safe here.

File: main/plugins/org.talend.designer.components.localprovider/components/tJMSInput/tJMSInput_finally.javajet (hunk 1 above)
Comment (vbogatyrov) [inline main/plugins/org.talend.designer.components.localprovider/components/tJMSInput/tJMSInput_finally.javajet]: Changed exception variable name to e_<%=cid%>

**Comment by vbogatyrov** [comment]
Switched from use of globalMap to resourceMap

**Comment by wwang-talend** [comment]
please create a QA issue for us to test it in component side.

**Pull request merged**